connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/mapping/CloudEventsMapper.java

```diff
@@ -0,0 +1,187 @@
+import org.eclipse.ditto.protocol.ProtocolFactory;
+
+/**
+ * A message mapper implementation for the Mapping incoming CloudEvents to Ditto Protocol.
+ */
+@PayloadMapper(alias = {"CloudEvents",
+    // legacy full qualified name
+    "org.eclipse.ditto.connectivity.service.mapping.CloudEventsMapper"})
+public final class CloudEventsMapper extends AbstractMessageMapper {
+
+
+  static final String CE_ID = "ce-id";
+  static final String CE_TYPE = "ce-type";
+  static final String CE_SOURCE = "ce-source";
+  static final String CE_SPECVERSION = "ce-specversion";
+
+  static final String contentType = "application/cloudevents+json";
+
+  // make all of them static
+  static final String SPECVERSION = "specversion";
+  static final String ID = "id";
+  static final String SOURCE = "source";
+  static final String TYPE = "type";
+  // declare uuid in the function
+  // make static variables uppercase
+
+
+  static final String OUTBOUNDTYPE = "org.eclipse.ditto.outbound";
+
+  static final String OUTBOUNDSPECVERSION = "1.0";
+
+  static final String OUTBOUNDSOURCE = "org.eclipse.ditto";
+
+
+  private String base64decoding(final String base64Message) {
+    byte[] messageByte = Base64.getDecoder().decode(base64Message);
+    String decodedString = new String(messageByte);
+    return decodedString;
+  }
+
+  boolean validatePayload(String payload) {
+    final JsonObject jsonObject = newObject(payload);
+    if (jsonObject.getValue(SPECVERSION).isPresent() && jsonObject.getValue(TYPE).isPresent()
+        && jsonObject.getValue(ID).isPresent() && jsonObject.getValue(SOURCE).isPresent()) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  private JsonifiableAdaptable extractData(final String message) throws Exception {
+    final JsonObject payloadJson = newObject(message);
+    if (payloadJson.getValue("data_base64").isPresent()) {
+      final String base64Data = payloadJson.getValue("data_base64").get().asString();
+      final String decodedData = base64decoding(base64Data);
+      final JsonifiableAdaptable decodedJsonifiableAdaptable = DittoJsonException.wrapJsonRuntimeException(
+          () -> ProtocolFactory.jsonifiableAdaptableFromJson(newObject(decodedData)));
+      return decodedJsonifiableAdaptable;
+    }
+    if (payloadJson.getValue("data").isPresent()) {
+      final String data = payloadJson.getValue("data").get().toString();
+      return DittoJsonException.wrapJsonRuntimeException(
+          () -> ProtocolFactory.jsonifiableAdaptableFromJson(newObject(data)));
+    } else {
+      throw new Exception("Invalid CloudEvent");
+    }
+  }
+
+  public boolean checkHeaders(final ExternalMessage message) {
+    Map<String, String> headers = message.getHeaders();
+    if (headers.get(CE_ID) == null || headers.get(CE_SOURCE) == null
+        || headers.get(CE_TYPE) == null || headers.get(CE_SPECVERSION) == null) {
+      return false;
+    } else {
+      return true;
+    }
```

Comment:
please use `containsKey` instead of checking for `null`.
Can also be simplified a lot to:
```suggestion
    final Map<String, String> headers = message.getHeaders();
    return message.getHeaders().containsKey(CE_ID) && 
            headers.containsKey(CE_SOURCE) && 
            headers.containsKey(CE_TYPE) && 
            headers.containsKey(CE_SPECVERSION);
```